aten/src/ATen/cudnn/Utils.h

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+#include <ATen/ATen.h>
+#include "THC/THC.h"
+#include "cudnn-wrapper.h"
+#include "Handles.h"
+
+namespace at { namespace cudnn {
+
+inline void cudnnSetStreamToCurrent() {
+  // TODO: Should we call lazyInitCUDA() or access thc_state directly?
+  // TODO: Should getCurrentStream be a method on Context?
+  cudnnSetStream(getCudnnHandle(), THCState_getCurrentStream(globalContext().lazyInitCUDA()));
+}
+
+// TODO: Move this out to ATen proper?
+inline Tensor contiguousIfZeroInStrides(const Tensor& t) {
+  for (auto s : t.strides()) {
+    if (s == 0) return t.contiguous();
```

Comment:
Whaaaaat? When is this function even used? It looks dangerous to me that you determine a tensor is contiguous based on a *single* stride value (having 0 stride means that this dim is contiguous, not whole tensor).

Reply:
This code was lifted from some existing code in the affine grid generator bindings. My best explanation of what is going on here is this comment (which should be added here):

```
            # Sometimes grad_output is a scalar (like 1) expanded as a tensor.
            # cudnn requires a tensor that has non-zero strides.
```

Reply:
I still don't get how does it explain anything. It's not a correct way to check that a tensor is an expanded scalar

Reply:
I think what's going on is cudnn has a buggy check for tensor being contiguous (that is, it does not ignore stride for dimension that is equal to 1). And this function does not determine if tensor is contiguous, it makes a tensor that has a zero stride contiguous (hopefully setting strides for size=1 to cudnn't liking). If that's indeed the case, it should be commented better.

Reply:
Commented.